src/back/TicketManager/src/PdfGenerator/TicketManager.PdfGenerator.Api/Processes/GenerateTicketPdfConsumer.cs

```diff
@@ -27,52 +31,18 @@ public Task Consume(ConsumeContext<GenerateTicketPdf> context)
         
         logger.LogInformation("Ticket {TicketId} pdf generation started", ticket.Id);
 
-        PdfWriter writer = new PdfWriter($"ticket-{ticket.Id}.pdf");
-        PdfDocument pdf = new PdfDocument(writer);
-        Document document = new Document(pdf);
-        
-        Paragraph paragraph = new Paragraph(@event.Name)
-            .SetTextAlignment(TextAlignment.CENTER)
-            .SetFontSize(32);
-        document.Add(paragraph);
-        
-        paragraph = new Paragraph($"{@event.Location}, {@event.Date:dd/MM/yyyy HH:mm}")
-            .SetTextAlignment(TextAlignment.CENTER)
-            .SetFontSize(20);
-        document.Add(paragraph);
-        
-        paragraph = new Paragraph(@event.Description)
-            .SetTextAlignment(TextAlignment.CENTER)
-            .SetFontSize(14);
-        document.Add(paragraph);
-        
-        paragraph = new Paragraph($"Organizer: {@event.Organizer.DisplayName}\n {@event.Organizer.PhoneNumber}, {@event.Organizer.Address}")
-            .SetTextAlignment(TextAlignment.CENTER)
-            .SetFontSize(12);
-        document.Add(paragraph);
-        document.Add(new Paragraph("\n"));
-        document.Add(new Paragraph("\n"));
-
-        paragraph = new Paragraph($"Ticket for {user.FirstName} {user.LastName}, born on {user.BirthDate:dd/MM/yyyy}")
-            .SetTextAlignment(TextAlignment.LEFT)
-            .SetFontSize(20);
-        document.Add(paragraph);
-
-        Table table = new Table(ticket.Seats.Count);
-        table.SetWidth(UnitValue.CreatePercentValue(60));
-        table.SetHorizontalAlignment(HorizontalAlignment.CENTER);
-        
-        foreach (var seat in ticket.Seats)
+        using (var stream = new MemoryStream())
         {
-            table.AddCell(new Cell()
-                .Add(new Paragraph($"Row: {seat.Row} Column: {seat.Column}")
-                    .SetTextAlignment(TextAlignment.JUSTIFIED)));
-            table.StartNewRow();
+            GeneratePdfFile.Generate(stream, ticket, @event, user);
+            stream.Position = 0;
+            
+            var blobServiceClient = new BlobServiceClient(blobConfiguration.ConnectionString);
```

Comment:
Generally you shouldn't create instance here, but inject it to constructor. But in this specific scenario we can live with it.